pilot/pkg/bootstrap/configcontroller.go

```diff
@@ -58,10 +63,17 @@ const (
 	requiredMCPCertCheckFreq = 500 * time.Millisecond
 )
 
+var useBuiltinGalleyConfigSource = env.RegisterBoolVar("USE_BUILTIN_GALLEY_SOURCE", false,
```

Comment:
nit:  maybe just `USE_EMBEDDED_GALLEY`?

Reply:
This needs to go into features. No adhoc env vars in code please.